client/flutter/lib/pages/onboarding/notifications_page.dart

```diff
@@ -25,14 +32,31 @@ class _NotificationsPageState extends State<NotificationsPage> {
   }
 
   void _allowNotifications() async {
+    // iOS needs a call for permissions but Android doesn't.
+    // If the user opts-out on Android we just don't register the device token.
+    if (Platform.isIOS) {
+      await _firebaseMessaging.requestNotificationPermissions(
+        const IosNotificationSettings(
+        sound: true, badge: true, alert: true, provisional: false));
+    }
+    
+    await UserPreferences().setNotificationsEnabled(true);
+    await _registerFCMToken();
     _complete();
   }
 
   void _skipNotifications() async {
+    await UserPreferences().setNotificationsEnabled(false);
     _complete();
   }
 
   void _complete() {
     widget.onNext();
   }
+
+  void _registerFCMToken() async {
+    await _firebaseMessaging.getToken().then((String token) {
```

Comment:
Not a blocker but since this is using await you could get the token as the return value instead of using `.then()`.  Just a stylistic change, not important.